deeprank2/trainer.py

```diff
@@ -739,6 +748,9 @@ def _eval( # pylint: disable=too-many-locals
                 loss_ = loss_func(pred, y)
                 count_predictions += pred.shape[0]
                 sum_of_losses += loss_.detach().item() * pred.shape[0]
+            else:
+                target_vals += ['None'] * pred.shape[0]
+                eval_loss = 'None'
```

Comment:
Why are these strings instead of a `None`? Same question below.

I also noticed that the `_eval` method returns this `eval_loss`, but that 3/4 times the method is called, it looks like the return value is not stored. 
I am not sure if the `OutputExporterCollection` is dealing with that as I am not perfectly clear on what the `with self._output_exporters` line does. If it does, I think it would be useful to document that in the code somehow, either in the docstring of `_eval` or as a comment in the code.

Finally, the typehint for the return value should be updated in the definition of the code to allow either type.

Reply:
Not storing the return value doesn't give particular issues, and it exists in the local scope of the methods only